sunpy/net/vso/vso.py

```diff
@@ -40,33 +40,63 @@
 from .table_response import VSOQueryResponseTable
 from .zeep_plugins import SunPyLoggingZeepPlugin
 
-DEFAULT_URL_PORT = [{'url': 'http://docs.virtualsolar.org/WSDL/VSOi_rpc_literal.wsdl',
-                     'port': 'nsoVSOi'},
-                    {'url': 'https://sdac.virtualsolar.org/API/VSOi_rpc_literal.wsdl',
-                     'port': 'sdacVSOi'}]
+DEFAULT_URL_PORT = [
+    {'url': 'http://docs.virtualsolar.org/WSDL/VSOi_rpc_literal.wsdl', 'port': 'nsoVSOi'},
+    {'url': 'https://sdac.virtualsolar.org/API/VSOi_rpc_literal.wsdl', 'port': 'sdacVSOi'},
+    # This connects to the same cgi as the first WSDL file, but if that file
+    # isn't accessible and the SDAC cgi is unreachable, this might still work.
+    {'url': 'https://sdac.virtualsolar.org/API/VSOi_rpc_literal.wsdl', 'port': 'nsoVSOi'},
+]
 
 
 class _Str(str):
-    """ Subclass of string that contains a meta attribute for the
-    record_item associated with the file. """
+    """
+    Subclass of string that contains a meta attribute for the
+    record_item associated with the file.
+    """
     meta = None
 
 
 # ----------------------------------------
 def check_connection(url):
     try:
-        return urlopen(url).getcode() == 200
+        return urlopen(url, timeout=15).getcode() == 200
     except (socket.error, socket.timeout, HTTPError, URLError) as e:
         warn_user(f"Connection to {url} failed with error {e}. Retrying with different url and port.")
         return None
 
 
+def check_cgi_connection(url):
+    """
+    At the moment there is no way to make an "are you alive" request to the
```

Comment:
Could you quickly document in the docstring what returning `None` means here, as opposed to `True` or `False`?

Reply:
I made it return False, I was just copying the previous implementation, but anything Falsey works.